cxx-sensors/src/main/java/org/sonar/cxx/sensors/functioncomplexity/CxxFunctionComplexitySquidSensor.java

```diff
@@ -52,9 +55,9 @@
 
   private int linesOfCodeOverThreshold;
 
-  private Hashtable<SourceFile, FunctionCount> complexFunctionsPerFile = new Hashtable<>();

```

Comment:
Can you avoid the synchronized Hashtable and use a HashMap instead (squid:S1149) for complexFunctionsPerFile , locInComplexFunctionsPerFile ?

Reply:
for now `org.sonar.squidbridge.AstScanner::scanFiles()` scans all files sequentially so synchronization is not necessary indeed. I will change Hashtable to HashMap, but whoever implements  #1493 must be aware of this change.